packages/datetime/src/dateInput.tsx

```diff
@@ -298,7 +298,9 @@ export class DateInput extends AbstractComponent<IDateInputProps, IDateInputStat
     private handleInputBlur = () => {
         const valueString = this.state.valueString;
         let value = moment(valueString, this.props.format);
-        if (valueString !== this.getDateString(this.state.value) && (!value.isValid() || !this.dateIsInRange(value))) {
+        if (valueString.length &&
```

Comment:
please avoid implicit type coercion to booleans. instead use `valueString.length > 0`